challenges/6-numeros-primos/php/gdiasb/primos.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+function primo(int $num)
+{
+    if ($num == 1 || $num <= 0)
+    {
+        return 'false';
+    } elseif ($num == 2)
+    {
+        return 'true';
+    } 
+    for ($i = 2; $i <= $num; $i++)
+    {
+        if ($num % $i == 0) 
+        {
+            return 'false';
+        } else
+        {
+            return 'true';
+        }
+    }
```

Comment:
Aqui vai uma sugestão que deve funcionar! Alguns números estavam com problemas, mas, ao invés de apenas responder que estava com problema resolvi te dar uma ideia do que fazer.

Nesse laço teoricamente você *não pode* retornar com um else dentro, porque senão sempre ele vai cair no else e retornar algo, independente de qual passo você estiver (utilize uma ferramenta de debug que você vai ver, sempre vai no else), assim, o `return 'true'` deve estar por fora.

Deixando por fora, sempre vai retornar `'false'`, porque no seu laço estava contabilizando o passo como `<=`, sendo que quando eles são iguais realmente eles são divisíveis (como 7, no passo 7/7 tem resto 0, afinal, números primos são divisíveis por 1 e por eles mesmos apenas).

Acredito que isso deve solucionar o problema e espero que tenha esclarecido e te auxiliado!

```suggestion
    for ($i = 2; $i < $num; $i++)
    {
        if ($num % $i == 0)
        {
            return 'false';
        }
    }
    return 'true';
```